Source/WebCore/page/Page.cpp

```diff
@@ -2787,6 +2785,12 @@ void Page::setHeaderHeight(int headerHeight)
     if (!renderView)
         return;
 
+    auto frameRect = frameView->frameRect();
+    frameRect.setHeight(frameRect.height() - m_headerHeight + headerHeight);
+
+    m_headerHeight = headerHeight;
+
+    frameView->setFrameRect(frameRect);
```

Comment:
This isn't right. The size of the FrameView is the size of its frame, not the size of the contents. I think setting this fixes your scrollbar bug as a side effect.

Reply:
Yes you are right. We do not need to call `FrameView::setFrameRect()`. All we need is its side effect. In fact we need to call only `FrameView::availableContentSizeChanged()`.